implementation/src/test/java/io/smallrye/config/ProfileConfigSourceInterceptorTest.java

```diff
@@ -220,6 +220,29 @@ void multipleProfiles() {
         assertEquals("5678", config.getRawValue("my.prop"));
     }
 
+    @Test
+    void multipleProfilesWithDynamicKey() {
+        SmallRyeConfig config = new SmallRyeConfigBuilder()
+                .withSources(config(SMALLRYE_CONFIG_PROFILE, "\"%common,prof\"", "config_ordinal", "1000"))
```

Comment:
We don't expect the profiles to be quoted or include the `%`. Profiles are usually set as:

```properties
smallrye.config.profile=prod
smallrye.config.profile=common,prod
smallrye.config.profile=one,two,three
```

The `%` is just to mark that the next name in the configuration property is the profile to be used, but it shouldn't be required when setting the profile. 

Did you find a situation where you need the profiles to be quoted?

Reply:
No, I didn't find any practical example that requires quoting profile names. I will remove this part.

Reply:
Removed :)

Reply:
> I believe a better fix may be provided in ProfileConfigSourceInteceptor by rewriting the multiple profile keys to be part of the active profile ... 

Looking at your comment, maybe I missed something. Why does `ProfileConfigSourceInteceptor` need to be changed?

Reply:
Consider:

```java
@ConfigMapping
interface MyConfig {
 Map<String, String> foo();
}
```

```properties
foo.bar=baz
%dev.foo.bar=baz
%common,dev.foo.bar=baz
```

In this case, because the second segment of `foo.bar` is dynamic (the `Map` key), we cannot query it directly. We need to iterate the list of property names and find the ones in the `Map` namespace of `foo`.

With profiles, I guess we could search the profile namespace too (or multiple ones like in this case). On the other hand, when we added profiles, to not explode the list of property names, or even confuse users with the added prefix, we rewrite profile configuration names in their non-profile version (active):

https://github.com/smallrye/smallrye-config/blob/554783076711a2ccc89897075b237abfd19ef61c/implementation/src/main/java/io/smallrye/config/ProfileConfigSourceInterceptor.java#L60-L68

But in the case of multiple profile configuration names, this is not working. I believe that rewriting the multiple property names in their non-profiled format should do the trick (as long as the name found has any of the profiles active).

Reply:
Hey @ejba, how are you? Are you still planning to work on this? Thanks!